afs/models.py

```diff
@@ -139,73 +134,64 @@ def upload_model(self,
+                raise TypeError(
                     'Type error, loss is float.'
                 )
             evaluation_result.update({'loss': loss})
 
         if not isinstance(model_path, str):
-            raise AssertionError('Type error, model_name  cannot convert to string')
+            raise TypeError('Type error, model_name  cannot convert to string')
 
         if not os.path.isfile(model_path):
-            raise AssertionError('File not found, model path is not exist.')
-        else:
-            model_path = model_path
-            if os.path.sep in model_path:
-                model_path = model_path.split(os.path.sep)[-1]
-
-            if len(model_path) > 42 or len(model_path) < 1:
-                raise AssertionError('Model name length  is upper limit 1-42 char')
-
-            pattern = re.compile(r'(?!.*[^a-zA-Z0-9-_.]).{1,42}')
-            match = pattern.match(model_path)
-            if match is None:
-                raise AssertionError('Model naming rule is only a-z, A-Z, 0-9, - and _ allowed.')
-
-        # Check file size
-        if os.path.getsize(model_path) > 2*(1024*1024*1024):
-            raise AssertionError('Model size is upper limit 2 GB.')
+            raise IOError('File not found, model path is not exist.')
 
         with open(model_path, 'rb') as f:
             model_file = BytesIO(f.read())
         model_file.seek(0)
 
-        if self.api_version in 'v1':
-            self.repo_id = self.switch_repo(model_path)
-            if not self.repo_id:
-                self.repo_id = self.create_model_repo(model_path)
-        elif self.api_version == 'v2':
-            # Check default repo_id
-            if not self.repo_id:
-                # Find model_repo_id from name
-                if model_repository_name:
-                    self.get_model_repo_id(model_repository_name)
-                    # If not found, create one
-                    if not self.repo_id:
-                        self.repo_id = self.create_model_repo(model_repository_name)
-                else:
-                    raise ValueError('Please enter model_repository_name')
+        # Check default repo_id
+        if not self.repo_id:
+            # Find model_repo_id from name
+            if model_repository_name:
+                self.get_model_repo_id(model_repository_name)
+                # If not found, create one
+                if not self.repo_id:
+                    self.repo_id = self.create_model_repo(model_repository_name)
+            else:
+                raise ValueError('Please enter model_repository_name')
+    
+        # Fetch tags apm_node 
+        PAI_DATA_DIR = os.getenv('PAI_DATA_DIR', None)
+        if PAI_DATA_DIR:
+            PAI_DATA_DIR = json.loads(PAI_DATA_DIR)
+            if 'data' in PAI_DATA_DIR:
+                data = PAI_DATA_DIR['data']
+                if 'machineIdList' in data:
+                    machineIdList = data['machineIdList'].pop(0)
```

Comment:
```suggestion
                    machineIdList = data['machineIdList'].pop(0, None)
```
Use `pop()` with default would be more safe.

Reply:
Give a default value to check.